### PR TITLE
Register fonts in importing application target

### PR DESCRIPTION
**Pull request by ciaranmul**
This change aims to:
- Provide a mechanism, available to application targets which import the library, to register the custom fonts
- Not rely on UIKit so as to make said mechanism available to MacOS, WidgetKit, etc.
- Protocol the class that performs the action so that if a more SwiftUI-native interface is made available we can migrate
- Lazily register fonts when requested

Usage:
- fonts will automatically be registered by the registrar when first accessed

### CODE REVIEW

File: Sources/MyStyles/FontsRegistrar.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+//
+//  File.swift
+//  
+//
+//  Created by Ciarán Mulholland on 12/10/2021.
+//
```
Comment (rwgrier) [inline Sources/MyStyles/FontsRegistrar.swift]: Very silly thing (especially for a POC), but we don't need this header "stuff" do we?

**Review by rwgrier — APPROVED** [review]
One super minor nitpick. But lgtm.

File: Sources/MyStyles/FontsRegistrar.swift (hunk 1 above)
Comment (ciaranmul) [reply Sources/MyStyles/FontsRegistrar.swift]: nah we don't, I've just been lazy lol

**Review by maloneranger — APPROVED** [review]
👍

**Pull request merged**